### ISSUE TITLE
Please backport 76cad51ba700 to 14.x

### ISSUE THREAD

**Issue by mgorny** [llvm/llvm-project#53707]
76cad51ba700233d6e3492eddcbb466b6adbc2eb is needed to fix standalone builds.

```
commit 76cad51ba700233d6e3492eddcbb466b6adbc2eb
Author: Jameson Nash <vtjnash@gmail.com>
Date:   2022-02-08 00:46:33 +0100

    replace clang LLVM_ENABLE_PLUGINS -> CLANG_PLUGIN_SUPPORT in tests
    
    Ensure CLANG_PLUGIN_SUPPORT is compatible with llvm_add_library.
    Fixes an issue noted in D111100.
    
    Differential Revision: https://reviews.llvm.org/D119199
```

@vtjnash, could you confirm that it's fine to backport?

**Comment by tstellar** [comment]
Are there still issues with this patch in main or is it ready to backport?

**Comment by vtjnash** [comment]
I am unaware of any issues with it, but will make additional fixes if anyone asks

**Comment by tstellar** [comment]
Merged: 6e7a837fdd55 a42f1f88f136

**Comment by vtjnash** [comment]
Could you add 9d59cfc67eadbe4b4088d70f8bbc61c96568d2f1 now also (@statham-arm)
```
commit 9d59cfc67eadbe4b4088d70f8bbc61c96568d2f1 (HEAD -> arcpatch-D119591, upstream/main)
Author: Jameson Nash <vtjnash@gmail.com>
Date:   Fri Feb 11 16:42:37 2022 -0500

    clang-analyzer plugins require LLVM_ENABLE_PLUGINS also
    
    The clang-analyzer plugins are not linked to a particular tool, so they
    can only be compiled if plugins are broadly supported. We could opt
    instead to decide whether to link them to specifically against clang or
    with undefined symbols, depending on the value of LLVM_ENABLE_PLUGINS,
    but we do not currently expect there to be a use case for that rather
    niche configuration.
    
    Differential Revision: https://reviews.llvm.org/D119591
```

**Comment by tstellar** [comment]
Merged: 6e7a837fdd55 a42f1f88f136 880717cc801d
